server.go

```diff
@@ -84,6 +84,7 @@ func (s *RPCServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	s.handleReader(ctx, r.Body, w, rpcError)
+	_ = r.Body.Close() // nolint
```

Comment:
```
    // The Server will close the request body. The ServeHTTP
    // Handler does not need to.
    Body io.ReadCloser
```

~https://golang.org/pkg/net/http/#Request

Reply:
If on the other hand, you're doing something funky and not using `http`, you might want to use defer since you guys are recovering from panics quite liberally in the code. Maybe not relevant here, just a memo.